pkg/destroy/openstack/openstack.go

```diff
@@ -97,52 +98,63 @@ func (o *ClusterUninstaller) Run(context.Context) error {
 		"deleteFloatingIPs":    deleteFloatingIPs,
 		"deleteImages":         deleteImages,
 	}
-	returnChannel := make(chan string)
 
 	opts := &clientconfig.ClientOpts{
 		Cloud: o.Cloud,
 	}
 
+	errChan := make(chan (error), len(deleteFuncs))
+
+	var wg sync.WaitGroup
+
 	// launch goroutines
-	for name, function := range deleteFuncs {
-		go deleteRunner(name, function, opts, o.Filter, o.Logger, returnChannel)
+	for n, f := range deleteFuncs {
+		wg.Add(1)
+		go func(name string, function deleteFunc) {
+			defer wg.Done()
+			err := deleteRunner(ctx, function, opts, o.Filter, o.Logger.WithField("deleteFuncName", name))
```

Comment:
is the `deleteFuncName` here to stay? The output feels a bit "loaded".

```
...
DEBUG Deleting openstack security-groups            deleteFuncName=deleteSecurityGroups
DEBUG Deleting openstack networks                   deleteFuncName=deleteNetworks
DEBUG Deleting Port "f5f45370-10ed-421c-82f9-7c771482acc7"  deleteFuncName=deletePorts
DEBUG Deleting Subnet: "87507b1d-6be9-4fee-a560-84d87ff3ae93"  deleteFuncName=deleteSubnets
DEBUG Deleting network: "95264d20-2c03-4f95-84c6-f6faa8e8c6af"  deleteFuncName=deleteNetworks
DEBUG Deleting Security Group: "04fbdf3c-f3e5-43b8-b9b0-801fb4d167ae"  deleteFuncName=deleteSecurityGroups
DEBUG Exiting deleting openstack ports              deleteFuncName=deletePorts
DEBUG Deleting Security Group: "ab3903e2-e659-4253-9445-70cf351a26e4"  deleteFuncName=deleteSecurityGroups
DEBUG Deleting openstack trunks                     deleteFuncName=deleteTrunks
DEBUG Exiting deleting openstack security-groups    deleteFuncName=deleteSecurityGroups
DEBUG Exiting deleting openstack subnets            deleteFuncName=deleteSubnets
...
```

Reply:
I am fine with removing it if you do not want it. I added it to preserve the info that had been logged in the "goroutine %v complete" debug entry.